### PR TITLE
Prep v0.14.0 release

### LINKED ISSUE #344: New v0.14 (pre-)release?
We just landed some pretty substantial new functionality in Tendermint KMS: transaction signing using a YubiHSM2-backed account key:

https://github.com/iqlusioninc/tmkms/pull/78

In the process we've updated to `tendermint-rs` HEAD including the `tendermint-rpc` crate split, which is very nice for this application, since we can leave that crate out and with it all of the associated Tokio and other network dependencies (we have the KMS automatically submit signed transactions via RPC). We have it all running in production successfully 🎉

Additionally, we really want #322 which updates Signatory and RustCrypto dependencies.

I'd like to cut another alpha release of the KMS with this, but right now I'm depending on the `tendermint` and `tendermint-rpc` crates as a git-based dependency.

Would it be possible to cut a v0.14.0(-pre/alpha) release that incorporates all of the latest changes? I'd be happy to put together a PR which does the relevant prep work.

### PR DESCRIPTION
**Pull request by Liamsi**
rendered changelog: https://github.com/informalsystems/tendermint-rs/blob/ismail/release/prep_v0.14.0/CHANGES.md#0140-2020-06-19

closes #344 

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

**Comment by Liamsi** [comment]
>  someone was too quick on the merge button. :p The PR is #343

I can be impatient, too :-P 

It's still in draft, I'll add this and:
 - [ ] mention the light client refactor 
 - [ ] add deprecation warning for `lite` module
 - [ ] short summary of changes (it looks like we've did a lot but there is not that much)

File: CHANGES.md [hunk 1]
```diff
@@ -1,33 +1,58 @@
-## Pending
+## [0.14.0] (2020-06-19)
+
+This release mainly targets compatibility with Tendermint [v0.33.x] but contains a lot of smaller improvements regarding testing and (de)serialization.
+Also noteworthy that the rpc module was broken out into a separate crate ([tendermint-rpc]).
```
Comment (Liamsi) [inline CHANGES.md:4]:
```suggestion
Also noteworthy is that the rpc module was broken out into a separate crate ([tendermint-rpc]).
```

File: tendermint/src/lib.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@
     unused_qualifications
 )]
 #![forbid(unsafe_code)]
-#![doc(html_root_url = "https://docs.rs/tendermint/0.13.0")]
+#![doc(html_root_url = "https://docs.rs/tendermint/0.14.0")]
```
Comment (Liamsi) [inline tendermint/src/lib.rs:16]: We should add this to sub-crates too:
https://doc.rust-lang.org/rustdoc/the-doc-attribute.html#html_root_url

File: light-client/Cargo.toml [hunk 3]
```diff
@@ -3,10 +3,11 @@ name = "tendermint-light-client"
 version = "0.1.0"
```
Comment (xla) [inline light-client/Cargo.toml:3]:
```suggestion
version = "0.14.0"
```

For consistency.

**Comment by tarcieri** [comment]
@xla can you (or someone else) publish `tendermint-rpc`?

**Comment by Liamsi** [comment]
working on it:
```
Finished dev [unoptimized + debuginfo] target(s) in 1m 10s
   Uploading tendermint-rpc v0.14.0 (/Users/ismail/IdeaProjects/tendermint-rs/rpc)
error: api errors (status 200 OK): missing or empty metadata fields: license. Please see https://doc.rust-lang.org/cargo/reference/manifest.html for how to upload metadata
```

**Pull request merged**